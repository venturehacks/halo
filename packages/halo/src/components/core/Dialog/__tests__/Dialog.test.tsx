import { mount, render, shallow } from 'enzyme';
import React from 'react';

import { Dialog } from '../index';

describe('Dialog', () => {
  test('smoke', () => {
    const component = render(<Dialog>content</Dialog>);
    expect(component.text()).toEqual('content');
  });

  describe('features', () => {
    describe('title', () => {
      test('string', () => {
        const component = render(<Dialog title="My title">content</Dialog>);
        expect(component.text()).toContain('My title');
      });

      test('JSX', () => {
        const jsxTitle = <span data-test="jsx">JSX Title</span>;
        const wrapper = shallow(<Dialog title={jsxTitle}>content</Dialog>);
        expect(wrapper.contains(jsxTitle)).toEqual(true);
      });
    });

    describe('button', () => {
      test('primary', () => {
        const callback = jest.fn();
        const wrapper = mount(
          <Dialog onPrimaryButtonClick={callback} primaryButtonLabel="OK">
            Demo
          </Dialog>,
        );

        wrapper
          .find(`[data-test="Dialog-primaryButton"]`)
          .last()
          .simulate('click');

        expect(callback).toBeCalledTimes(1);
      });

      test('secondary', () => {
        const callback = jest.fn();
        const wrapper = mount(
          <Dialog
            onSecondaryButtonClick={callback}
            secondaryButtonLabel="Cancel"
          >
            Demo
          </Dialog>,
        );

        wrapper
          .find(`[data-test="Dialog-secondaryButton"]`)
          .last()
          .simulate('click');

        expect(callback).toBeCalledTimes(1);
      });
    });
  });

  describe('snapshot', () => {
    test('default', () => {
      const component = render(<Dialog>content</Dialog>);
      expect(component).toMatchSnapshot();
    });

    test('inline context', () => {
      const component = render(<Dialog context="inline">content</Dialog>);
      expect(component).toMatchSnapshot();
    });

    test('no content padding', () => {
      const component = render(<Dialog contentPadding={false}>content</Dialog>);
      expect(component).toMatchSnapshot();
    });

    test('title + both buttons', () => {
      const component = render(
        <Dialog
          primaryButtonLabel="Save"
          secondaryButtonLabel="Cancel"
          title="Title"
        >
          content
        </Dialog>,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
