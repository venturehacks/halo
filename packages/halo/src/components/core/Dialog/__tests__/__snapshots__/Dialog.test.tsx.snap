// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog snapshot default 1`] = `
<div
  class="component contextModal paddingDefault"
  data-test="Dialog"
>
  <div
    class="content"
  >
    content
  </div>
</div>
`;

exports[`Dialog snapshot inline context 1`] = `
<div
  class="component contextInline paddingDefault"
  data-test="Dialog"
>
  <div
    class="content"
  >
    content
  </div>
</div>
`;

exports[`Dialog snapshot no content padding 1`] = `
<div
  class="component contextModal"
  data-test="Dialog"
>
  <div
    class="content"
  >
    content
  </div>
</div>
`;

exports[`Dialog snapshot title + both buttons 1`] = `
<div
  class="component contextModal paddingDefault"
  data-test="Dialog"
>
  <div
    class="title"
  >
    <h4
      class="component __halo_fontSizeMap_size--lg __halo_fontWeight_medium"
    >
      Title
    </h4>
  </div>
  <div
    class="content"
  >
    content
  </div>
  <div
    class="chin"
    data-test="Dialog-chin"
  >
    <button
      class="component secondary sm emphasis"
      data-test="Dialog-secondaryButton"
      type="button"
    >
      Cancel
    </button>
    <button
      class="component primary sm emphasis"
      data-test="Dialog-primaryButton"
      type="button"
    >
      Save
    </button>
  </div>
</div>
`;
